components/wmtk_components/embedded_component/internal/EmbeddedRemeshing.cpp

```diff
@@ -0,0 +1,117 @@
+#include "EmbeddedRemeshing.hpp"
+
+namespace wmtk::components::internal {
+EmbeddedRemeshing2D::EmbeddedRemeshing2D(
+    Eigen::MatrixXi& E_,
+    Eigen::MatrixXd& V_,
+    double blank_rate_,
+    double resolute_area_)
+    : E(E_)
+    , V(V_)
+    , blank_rate(blank_rate_)
+    , resolute_area(resolute_area_)
+{
+    F.resize(0, 0);
+    for (int i = 0; i < V.rows(); ++i) {
+        markedV.push_back(i);
+    }
+    for (int i = 0; i < E.rows(); ++i) {
+        markedE.push_back(std::pair<int, int>(E(i, 0), E(i, 1)));
+    }
+}
+void EmbeddedRemeshing2D::compute_bounding_vaule(
+    double& max_x,
+    double& max_y,
+    double& min_x,
+    double& min_y)
+{
+    max_x = max_y = std::numeric_limits<double>::lowest();
+    min_x = min_y = std::numeric_limits<double>::max();
+    for (int i = 0; i < V.rows(); ++i) {
+        max_x = std::max(max_x, V(i, 0));
+        max_y = std::max(max_y, V(i, 1));
+        min_x = std::min(max_x, V(i, 0));
+        min_y = std::min(max_x, V(i, 1));
+    }
+    double blank_region_length_x = (max_x - min_x) * blank_rate;
+    double blank_region_length_y = (max_y - min_y) * blank_rate;
+    max_x += blank_region_length_x;
+    min_x -= blank_region_length_x;
+    max_y += blank_region_length_y;
+    min_y -= blank_region_length_y;
+}
+void EmbeddedRemeshing2D::process()
+{
+    // find boundingbox
+    double max_x, min_x, max_y, min_y;
+    compute_bounding_vaule(max_x, max_y, min_x, min_y);
+
+    Eigen::MatrixXd BoundingBoxV(4, 2);
+    Eigen::MatrixXi BoundingBoxE(4, 2);
+    BoundingBoxV << max_x, max_y, min_x, max_y, min_x, min_y, max_x, min_y;
+    BoundingBoxE << markedV.size(), markedV.size() + 1, markedV.size() + 1, markedV.size() + 2,
+        markedV.size() + 2, markedV.size() + 3, markedV.size() + 3, markedV.size();
+    Eigen::MatrixXd tempV;
+    Eigen::MatrixXi tempE;
+    tempV = V + BoundingBoxV;
+    tempE = E + BoundingBoxE;
+
+    Eigen::MatrixXd H;
+    H.resize(1, 2);
+    H << max_x + 1.0, max_y + 1.0;
+
+    // the fourth parameter is resolution, we should discuss it maybe.
+    igl::triangle::triangulate(tempV, tempE, H, "a0.1q", V, F);
+
+    // need to connect the topology
```

Comment:
I am not entirely sure what the code below does but I think this component should return after triangulating.

Reply:
this block is important since the igl::triangulate function will automatically split edges during the triangulating process, which means the input vertices will not connect to each other anymore. So I need to do some postprocessing to guarantee the input will connect to each other even they have been splitted.